Content.Shared/GameObjects/Components/Disposal/SharedDisposableComponent.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using Content.Shared.GameObjects.EntitySystems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Serialization;
+
+namespace Content.Shared.GameObjects.Components.Disposal
+{
+    public abstract class SharedDisposableComponent : Component, IActionBlocker
+    {
+        public override string Name => "Disposable";
+
+        public override uint? NetID => ContentNetIDs.DISPOSABLE;
+
+        public abstract bool InTube { get; protected set; }
+
+        bool IActionBlocker.CanInteract()
```

Comment:
In SS13 you can totally still do interactions while inside a disposals tube. Though obviously you cannot interact with the outside world much. It's probably best if you remove the interaction blocking (thus removing the need for this component altogether) and modified interaction system to disallow interacting outside your current container?